src/client/sandbox/xhr.js

```diff
@@ -160,5 +158,20 @@ export default class XhrSandbox extends SandboxBase {
 
             return nativeMethods.xhrSetRequestHeader.call(this, header, value);
         };
+
+        overrideDescriptor(window.XMLHttpRequest.prototype, 'status', function () {
+            const status = nativeMethods.xhrStatusGetter.call(this);
+
+            return this[IS_OPENED_XHR] && status === SAME_ORIGIN_CHECK_FAILED_STATUS_CODE ? 0 : status;
```

Comment:
I guess the condition `this[IS_OPENED_XHR]` is not necessary.
And constant too.